### PR TITLE
ISSUES-369 add default database when create materialize view

### LINKED ISSUE #369: StorageView: Could not retrieve database name from select query.
So, we have two MergeTree tables:

```sql
CREATE TABLE default.test1
(
    date Date, 
    platform Enum8('a' = 0, 'b' = 1, 'c' = 2), 
    app Enum8('a' = 0, 'b' = 1)
) ENGINE = MergeTree(date, (platform, app), 8192)
```
and
```sql
CREATE TABLE default.test2
(
    date Date, 
    platform Enum8('a' = 0, 'b' = 1, 'c' = 2), 
    app Enum8('a' = 0, 'b' = 1)
) ENGINE = MergeTree(date, (platform, app), 8192)
```

And then
```sql
CREATE TABLE groups
(
    platform Enum8('a' = 0, 'b' = 1, 'c' = 2), 
    app Enum8('a' = 0, 'b' = 1), 
    group Enum8('AA' = 0, 'AB' = 1, 'BA' = 2, 'BB' = 3, 'CA' = 4, 'CB' = 5)
) ENGINE = Join(ALL, INNER, platform, app)
```

Now, joining works:
```sql
SELECT 
    date, 
    platform, 
    app, 
    group
FROM 
(
    SELECT *
    FROM test1 
    UNION ALL 
    SELECT *
    FROM test2 
) 
ALL INNER JOIN groups USING (platform, app)
```

But we can't create materialized view upon it:
```sql
CREATE MATERIALIZED VIEW bla ENGINE = MergeTree(date, (platform, app), 8192) POPULATE AS
SELECT 
    date, 
    platform, 
    app, 
    group
FROM 
(
    SELECT *
    FROM test1 
    UNION ALL 
    SELECT *
    FROM test2 
) 
ALL INNER JOIN groups USING (platform, app)
```
It says
```
Received exception from server:
Code: 49. DB::Exception: Received from localhost:9000, 127.0.0.1. DB::Exception: Logical error while creating StorageView. Could not retrieve database name from select query..
```
We can specify **default.** for either test1 or test2 (or, obviously, for both) and this solves the issue:
```sql
CREATE MATERIALIZED VIEW bla ENGINE = MergeTree(date, (platform, app), 8192) POPULATE AS
SELECT 
    date, 
    platform, 
    app, 
    group
FROM 
(
    SELECT *
    FROM default.test1 
    UNION ALL 
    SELECT *
    FROM test2 
) 
ALL INNER JOIN groups USING (platform, app)
```
I don't find this way nice though and It seems like a bug for me.
```
$ apt-cache policy clickhouse-server-common 
clickhouse-server-common:
  Installed: 1.1.54133
  Candidate: 1.1.54134
  Version table:
     1.1.54134 500
        500 http://repo.yandex.ru/clickhouse/xenial stable/main amd64 Packages
 *** 1.1.54133 100
        100 /var/lib/dpkg/status
```

It happens using both clickhouse-client and HTTP interface, so it rather looks like server-specific.

### PR DESCRIPTION
**Pull request by zhang2014**
fix #369
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

### CODE REVIEW

File: dbms/src/Storages/StorageMaterializedView.cpp [hunk 1]
```diff
@@ -76,6 +73,8 @@ StorageMaterializedView::StorageMaterializedView(
             "You must specify where to save results of a MaterializedView query: either ENGINE or an existing table in a TO clause",
             ErrorCodes::INCORRECT_QUERY);
 
+    /// Default value, if only table name exist in the query
+    select_database_name = local_context.getCurrentDatabase();
```
Comment (zhang2014) [inline dbms/src/Storages/StorageMaterializedView.cpp]: Encountered an error in restart.

**Comment by zhang2014** [comment]
depend #1984

**Comment by alexey-milovidov** [comment]
Will the modified query with explicitly specified database be written into `.sql` metadata file?
I suspect it won't. In that case the view will behave differently after server restart.

**Comment by zhang2014** [comment]
@alexey-milovidov I've added test, Please help review the changes : )

**Pull request merged**